autolens/analysis/analysis/dataset.py

```diff
@@ -0,0 +1,193 @@
+            dataset=dataset,
+            adapt_image_maker=adapt_image_maker,
+            cosmology=cosmology,
+            settings_inversion=settings_inversion,
+        )
+
+        AnalysisLens.__init__(
+            self=self,
+            positions_likelihood=positions_likelihood,
+            cosmology=cosmology,
+        )
+
+        self.preloads = self.preloads_cls()
+
+        self.raise_inversion_positions_likelihood_exception = (
+            raise_inversion_positions_likelihood_exception
+        )
+
+        if os.environ.get("PYAUTOFIT_TEST_MODE") == "1":
+            self.raise_inversion_positions_likelihood_exception = False
+
+    def modify_before_fit(self, paths: af.DirectoryPaths, model: af.Collection):
+        """
+        This function is called immediately before the non-linear search begins and performs final tasks and checks
+        before it begins.
+
+        This function:
+
+        - Checks that the adapt-dataset is consistent with previous adapt-datasets if the model-fit is being
+          resumed from a previous run.
+
+        - Checks the model and raises exceptions if certain critieria are not met.
+
+        Once inherited from it also visualizes objects which do not change throughout the model fit like the dataset.
+
+        Parameters
+        ----------
+        paths
+            The PyAutoFit paths object which manages all paths, e.g. where the non-linear search outputs are stored,
+            visualization and the pickled objects used by the aggregator output by this function.
+        model
+            The PyAutoFit model object, which includes model components representing the galaxies that are fitted to
+            the imaging data.
+        """
+
+        super().modify_before_fit(paths=paths, model=model)
+
+        self.raise_exceptions(model=model)
+
+    def raise_exceptions(self, model):
+        has_pix = model.has_model(cls=(aa.Pixelization,)) or model.has_instance(
+            cls=(aa.Pixelization,)
+        )
+
+        if has_pix:
+            if (
+                self.positions_likelihood is None
+                and self.raise_inversion_positions_likelihood_exception
+                and not conf.instance["general"]["test"][
+                    "disable_positions_lh_inversion_check"
+                ]
+            ):
+                raise exc.AnalysisException(
+                    "You have begun a model-fit which reconstructs the source using a pixelization.\n\n"
+                    "However, you have not input a `positions_likelihood` object.\n\n"
+                    "It is likely your model-fit will infer an inaccurate solution.\n\n "
+                    ""
+                    "Please read the following readthedocs page for a description of why this is, and how to set up"
+                    "a positions likelihood object:\n\n"
+                    ""
+                    "https://pyautolens.readthedocs.io/en/latest/general/demagnified_solutions.html"

```

Comment:
You can use a triple-quote to avoid having to pass formatting characters as Python interprets its contents literally (although you might not like the indentation)

```python
raise exc.AnalysisException(
                    """You have begun a model-fit which reconstructs the source using a pixelization.

However, you have not input a `positions_likelihood` object.

It is likely your model-fit will infer an inaccurate solution.

Please read the following readthedocs page for a description of why this is, and how to set up a positions likelihood object:
            
https://pyautolens.readthedocs.io/en/latest/general/demagnified_solutions.html"""
                )